com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/AnimationSetBuilder.java

```diff
@@ -110,35 +117,162 @@ private void buildAnimations(Task<Void> task, AnimationSetDesc.Builder animSetDe
+
             } catch (XMLStreamException e) {
                 throw new CompileExceptionError(animFile, e.getLocation().getLineNumber(), "Failed to load animation: " + e.getLocalizedMessage(), e);
             } catch (LoaderException e) {
                 throw new CompileExceptionError(animFile, -1, "Failed to load animation: " + e.getLocalizedMessage(), e);
             }
 
             animationSetBuilder.addAllAnimations(animBuilder.getAnimationsList());
-            if(animBuilder.getBoneListList().size() > boneList.size()) {
-                boneList = animBuilder.getBoneListList();
+        }
+    }
+
+// For the Editor
+
+    static public AnimationSetDesc getAnimationSetDesc(InputStream animationSetStream) throws IOException {
+        InputStreamReader animSetDescISR = new InputStreamReader(animationSetStream);
+        AnimationSetDesc.Builder animSetDescBuilder = AnimationSetDesc.newBuilder();
+        TextFormat.merge(animSetDescISR, animSetDescBuilder);
+        return animSetDescBuilder.build();
+    }
+
+    private static ArrayList<String> makeUnique(List<String> items) {
+        ArrayList<String> unique = new ArrayList<>();
+        for (String item : items) {
+            if (!unique.contains(item)) {
+                unique.add(item);
+            }
+        }
+        return unique;
+    }
+
+    static public ArrayList<String> getAnimationsPaths(AnimationSetDesc desc) {
+        ArrayList<String> animations = new ArrayList<String>();
+        for(AnimationInstanceDesc instance : desc.getAnimationsList()) {
+            animations.add(instance.getAnimation());
+        }
+        return makeUnique(animations);
+    }
+
+    static void loadColladaAnimations(AnimationSet.Builder animationSetBuilder, InputStream is, String animId, String parentId)
+    throws IOException, XMLStreamException, LoaderException {
+        ArrayList<String> localAnimationIds = new ArrayList<String>();
+        AnimationSet.Builder animBuilder = AnimationSet.newBuilder();
+        ColladaUtil.loadAnimations(is, animBuilder, animId, localAnimationIds);
+
+        animationSetBuilder.addAllAnimations(animBuilder.getAnimationsList());
+    }
+
+    static void loadModelAnimations(ArrayList<ModelImporter.Bone> bones, AnimationSet.Builder animationSetBuilder, InputStream is, String animId, String parentId,
+                                                            String path, ArrayList<String> animationIds) throws IOException {
+
+        System.out.printf("  modelutil anim: %s  parent: %s\n", animId, parentId);
+
+        ModelImporter.Scene scene = ModelUtil.loadScene(is, path, new ModelImporter.Options());
+
+        ArrayList<String> localAnimationIds = new ArrayList<String>();
+        AnimationSet.Builder animBuilder = AnimationSet.newBuilder();
+
+        // Currently, by design choice, each file must only contain one animation.
```

Comment:
Is this something we want to change later on?

Reply:
Not sure, but I don't think so.
We would have to change a lot of assumptions regarding the files+naming.